src/ui.js

```diff
@@ -30,108 +30,309 @@ const mdComponents = {
+    }
+
+    // Since we don't want to force this function to always be async, as it's only
+    // needed to be async for this syntax highlighting call, we will instead return
+    // an async function that can awaited on
+    return async () => {
+      await Promise.all(promises);
+      return domHTMLFragment;
+    };
+  } else {
+    // We aren't preforming any syntax highlighting, so lets return our rendered
+    // text.
+    return rendered;
+  }
+}
+
+/**
+ * @function buildRenderer
+ * @memberof markdown
+ * @desc Returns a Markdown Renderer instance with the provided options.
+ * Helpful to avoid having to build a new one over and over.
+ * @param {object} givenOpts - The optional arguments
  * @param {boolean} givenOpts.html - Whether HTML tags should be allowed.
- * @param {boolean} givenOpts.sanitize - If the page content should be saniized via DOMPurify.
- * @param {boolean} givenOpts.sanitizeAllowUnknownProtocols - Controls DOMPurify's
- * own option of 'ALLOW_UNKNOWN_PROTOCOLS'.
- * @param {boolean} givenOpts.sanitizeAllowSelfClose - Controls DOMPurify's
- * own option of 'ALLOW_SELF_CLOSE'
  * @param {boolean} givenOpts.breaks - If newlines should always be converted
  * into breaklines.
- * @param {boolean} givenOpts.handleFrontMatter - Whether frontmatter data should
- * processed and displayed.
- * @param {boolean} givenOpts.useDefaultEmoji - Whether `markdown-it-emoji` should be enabled.
- * @param {boolean} givenOpts.useGitHubHeadings - Whether `markdown-it-github-headings`
- * should be enabled. False by default.
- * @param {boolean} givenOpts.useTaskCheckbox - Whether `markdown-it-task-checkbox`
- * should be enabled. True by default.
+ * @param {boolean} givenOpts.emoji - If emojis should be included.
+ * @param {boolean} givenOpts.githubHeadings - Whether `markdown-it-github-headings`
+ * should be enabled.
+ * @param {boolean} givenOpts.taskCheckbox - Whether `markdown-it-task-checkbox`
+ * should be enabled.
  * @param {boolean} givenOpts.taskCheckboxDisabled - Controls `markdown-it-task-checkbox`
- * `disabled` option. True by default.
- * @param {boolean} givenOpts.taskCheckboxDivWrap - Controls `markdown-it-task-checkboc`
- * `divWrap` option. False by default.
- * @param {boolean} givenOpts.transformImageLinks - Attempt to resolve image URLs.
- * True by default.
- * @param {boolean} givenOpts.transformAtomLinks - Attempt to resolve links
- * pointing to Atom. True by Default.
- * @param {boolean} givenOpts.transformNonFqdnLinks - Attempt to resolve links
- * that are not fully qualified domain names. True by Default.
+ * `disabled` option.
+ * @param {boolean} givenOpts.taskCheckboxDivWrap - Controls `markdown-it-task-checkbox`
+ * `divWrap` option.
+ * @param {boolean} givenOpts.transformImageLinks - If links to images should
+ * attempted to be resolved.
+ * @param {boolean} givenOpts.transformNonFqdnLinks - If non-FQDN links should be
+ * resolved.
```

Comment:
```suggestion
 * @param {boolean} givenOpts.transformNonFqdnLinks - If non fully qualified
 * domain name links should be resolved.
```

This one is to avoid confusion with non-native English speakers :)